src/test/java/org/mozilla/zest/test/v1/ZestRequestUnitTest.java

```diff
@@ -42,6 +46,39 @@ public void testTokenReplacement() throws Exception {
 		assertEquals ("http://www.example.com/app/DEFG", req2.getUrl().toString());
 		assertEquals ("Set-Cookie: test=GHI", req2.getHeaders());
 		assertEquals ("test=JKL&user=12JKL34", req2.getData());
+		assertEquals ("ABC.JKL", req2.getZestCookies().get(0).getDomain());
+		assertEquals ("12JKL34", req2.getZestCookies().get(0).getName());
+		assertEquals ("56GHI78", req2.getZestCookies().get(0).getValue());
+		assertEquals ("/DEFGA/GHIB", req2.getZestCookies().get(0).getPath());
 	}
 
+	/**
+	 * Method testDeepCopy.
+	 * @throws Exception
+	 */
+	@Test
+	public void testDeepCopy() throws Exception {
+		ZestRequest req = new ZestRequest ();
+
+		req.setUrl(new URL("http://www.example.com/app/{{token1}}"));
+		req.setHeaders("Set-Cookie: test={{token2}}");
+		req.setData("test={{token3}}&user=12{{token3}}34");
+		req.addCookie(new ZestCookie("{{token}}.{{token3}}", "12{{token3}}34", "56{{token2}}78", "/{{token1}}A/{{token2}}B", new Date(), true));
+
+		ZestRequest req2 = req.deepCopy();
+
+		assertTrue(req.getHeaders().equals(req2.getHeaders()));
+		assertTrue(req.getData().equals(req2.getData()));
+
+		ZestCookie cookie = req.getZestCookies().get(0);
+		ZestCookie cookie2 = req2.getZestCookies().get(0);
+
+		assertTrue("cookies should not be reference equals", !cookie.equals(cookie2));
```

Comment:
Probably better to already use `!=`, otherwise it will start to fail (while it shouldn't) when `ZestCookie` properly implements `equals`.